### PR TITLE
Axis highlighted labels

### PR DESCRIPTION
**Pull request by antongreenely**
It's now possible to highlight an axis value (JUN on the screen shot)

### CODE REVIEW

File: Source/Charts/Charts/ChartViewBase.swift [hunk 1]
```diff
@@ -497,6 +497,22 @@ open class ChartViewBase: NSUIView, ChartDataProvider, AnimatorDelegate
...
+
+        if let xValue = h?.x
+        {
+            if let xValueIndex = xAxis.entries.index(of: xValue)
+            {
+                xAxis.selectedItemIndex = xValueIndex
+            }
+            else
+            {
+                xAxis.selectedItemIndex = NSInteger.min
```
Comment (GeorgeGreenely) [inline Source/Charts/Charts/ChartViewBase.swift]: Why? If no value is selected or you can't get an entry for it, it should not select anything

File: Source/Charts/Charts/ChartViewBase.swift [hunk 2]
```diff
@@ -497,6 +497,22 @@ open class ChartViewBase: NSUIView, ChartDataProvider, AnimatorDelegate
...
+            }
+        }
+        else
+        {
+            xAxis.selectedItemIndex = NSInteger.min
```
Comment (GeorgeGreenely) [inline Source/Charts/Charts/ChartViewBase.swift]: Why? If no value is selected or you can't get an entry for it, it should not select anything

File: Source/Charts/Components/XAxis.swift [hunk 3]
```diff
@@ -60,6 +60,16 @@ open class XAxis: AxisBase
     /// 
     /// **default**: 1.0
     @objc open var wordWrapWidthPercent: CGFloat = 1.0
+
+    /// index of selected entry on the axis
+    ///
+    /// **default**: NSInteger.min
+    @objc open var selectedItemIndex: NSInteger = NSInteger.min
```
Comment (GeorgeGreenely) [inline Source/Charts/Components/XAxis.swift]: Why have a default value? this should be an optional, if no item is selected then the value should be `nil` not `NSInteger.min`

Comment (antongreenely) [reply Source/Charts/Components/XAxis.swift]: NSInteger can't be nil as I understand. Do you recommend to use NSNumber instead?

Comment (GeorgeGreenely) [reply Source/Charts/Components/XAxis.swift]:
```
    /// index of selected entry on the axis
    ///
    /// **default**: nil
    open var selectedItemIndex: NSInteger? = nil
```

Comment (antongreenely) [reply Source/Charts/Components/XAxis.swift]: Will it work for ObjectiveC?

Comment (GeorgeGreenely) [reply Source/Charts/Components/XAxis.swift]: I guess not, but there has been places in the charts library where they've done that.

**Pull request merged**